### ISSUE TITLE
PC freezes when change img_mode into GRAY

### ISSUE THREAD

**Issue by wolfryu** [WenmuZhou/DBNet.pytorch#43]
RGB trains find.
but when change to GRAY, PC freezes, mem 100%.
here are the logs:
~~~
2020-06-24 15:42:53,301 DBNet.pytorch INFO: {'arch': {'backbone': {'in_channels': 1,
                       'pretrained': False,
                       'type': 'resnet18'},
          'head': {'k': 50, 'out_channels': 2, 'type': 'DBHead'},
          'neck': {'inner_channels': 256, 'type': 'FPN'},
          'type': 'Model'},
 'dataset': {'train': {'dataset': {'args': {'data_path': ['./datasets/train.txt'],
                                            'filter_keys': ['img_path',
                                                            'img_name',
                                                            'text_polys',
                                                            'texts',
                                                            'ignore_tags',
                                                            'shape'],
                                            'ignore_tags': ['*', '###'],
                                            'img_mode': 'GRAY',
                                            'pre_processes': [{'args': [{'args': {'p': 0.5},
                                                                         'type': 'Fliplr'},
                                                                        {'args': {'rotate': [-10,
                                                                                             10]},
                                                                         'type': 'Affine'},
                                                                        {'args': {'size': [0.5,
                                                                                           3]},
                                                                         'type': 'Resize'}],
                                                               'type': 'IaaAugment'},
                                                              {'args': {'keep_ratio': True,
                                                                        'max_tries': 50,
                                                                        'size': [640,
                                                                                 640]},
                                                               'type': 'EastRandomCropData'},
                                                              {'args': {'shrink_ratio': 0.4,
                                                                        'thresh_max': 0.7,
                                                                        'thresh_min': 0.3},
                                                               'type': 'MakeBorderMap'},
                                                              {'args': {'min_text_size': 8,
                                                                        'shrink_ratio': 0.4},
                                                               'type': 'MakeShrinkMap'}],
                                            'transforms': [{'args': {},
                                                            'type': 'ToTensor'},
                                                           {'args': {'mean': [0.485,
                                                                              0.456,
                                                                              0.406],
                                                                     'std': [0.229,
                                                                             0.224,
                                                                             0.225]},
                                                            'type': 'Normalize'}]},
                                   'type': 'ICDAR2015Dataset'},
                       'loader': {'batch_size': 1,
                                  'collate_fn': '',
                                  'num_workers': 6,
                                  'pin_memory': True,
                                  'shuffle': True}},
             'validate': {'dataset': {'args': {'data_path': ['./datasets/test.txt'],
                                               'filter_keys': [],
                                               'ignore_tags': ['*', '###'],
                                               'img_mode': 'GRAY',
                                               'pre_processes': [{'args': {'resize_text_polys': False,
                                                                           'short_size': 736},
                                                                  'type': 'ResizeShortSize'}],
                                               'transforms': [{'args': {},
                                                               'type': 'ToTensor'},
                                                              {'args': {'mean': [0.485,
                                                                                 0.456,
                                                                                 0.406],
                                                                        'std': [0.229,
                                                                                0.224,
                                                                                0.225]},
                                                               'type': 'Normalize'}]},
                                      'type': 'ICDAR2015Dataset'},
                          'loader': {'batch_size': 1,
                                     'collate_fn': 'ICDARCollectFN',
                                     'num_workers': 6,
                                     'pin_memory': False,
                                     'shuffle': True}}},
 'distributed': False,
 'local_rank': 0,
 'loss': {'alpha': 1, 'beta': 10, 'ohem_ratio': 3},
 'lr_scheduler': {'args': {'warmup_epoch': 3}, 'type': 'WarmupPolyLR'},
 'metric': {'args': {'is_output_polygon': False}, 'type': 'QuadMetric'},
 'name': 'DBNet_resnet18_FPN_DBHead',
 'optimizer': {'args': {'amsgrad': True, 'lr': 0.001, 'weight_decay': 0},
               'type': 'Adam'},
 'post_processing': {'args': {'box_thresh': 0.7,
                              'max_candidates': 1000,
                              'thresh': 0.3,
                              'unclip_ratio': 1.5},
                     'type': 'SegDetectorRepresenter'},
 'trainer': {'epochs': 1200,
             'finetune_checkpoint': '',
             'log_iter': 10,
             'output_dir': '/media/1234/12/3213/dbnetpytorch/DBNet.pytorch/output',
             'resume_checkpoint': '',
             'seed': 2,
             'show_images_iter': 50,
             'tensorboard': True}}
2020-06-24 15:42:53,332 DBNet.pytorch INFO: train with device cuda and pytorch 1.4.0
2020-06-24 15:42:57,967 DBNet.pytorch INFO: train dataset has 5979 samples,5979 in dataloader, validate dataset has 300 samples,300 in dataloader
~~~

**Comment by WenmuZhou** [comment]
I think there are a bug in base_dataset.py

remove try except code in https://github.com/WenmuZhou/DBNet.pytorch/blob/master/base/base_dataset.py#L53-L74 and try

**Comment by wolfryu** [comment]
it works! thank you!
